src/vm/eetwain.cpp

```diff
@@ -2659,8 +2664,14 @@ unsigned scanArgRegTableI(PTR_CBYTE     table,
 
                 if (hasPartialArgInfo)
                 {
+                    // We always leave argHigh pointing to the next ptr arg.
```

Comment:
Is this how it always was, or was this changed with RyuJIT?

Reply:
I am not very familiar with this code, and had to do a fair bit of spelunking to figure out how this is supposed to work, but looking at the code above, it assumes that argHigh is always pointing to the next ptr arg. The code has been like this since at least 2013, and I'm guessing that this particular case (partially interruptible ebp frame, nested calls, and non-ptr arg(s) below the last popped arg of the inner call) is rare enough that this hasn't been hit.

Reply:
That said, my plan was to do some gcstress testing on this, and then get reviews from people who might be more familiar with this code.

Reply:
I am familiar with the x86 GC encodings.

Reply:
@briansull - that's great! Could you have a look?
As I suspected, for this same test on desktop, jit32 un-nests the calls, so that it doesn't run into this issue.

Reply:
Oh, and in addition to the fix, it would be great if you could look at the comments I added and verify that they are correct. Figuring this out would have been a lot easier with more comments.

Reply:
All of the code dealing with `partialArgInfo `was recently introduced on the CoreCLR side of the world:

```
19b1ce56f95 (Jonghyun Park   2017-04-11 07:49:08 +0900 2435)     bool      hasPartialArgInfo;
42924c80d75 (Jonghyun Park   2017-04-05 09:30:08 +0900 2436) 
42924c80d75 (Jonghyun Park   2017-04-05 09:30:08 +0900 2437) #ifndef UNIX_X86_ABI
19b1ce56f95 (Jonghyun Park   2017-04-11 07:49:08 +0900 2438)     hasPartialArgInfo = info->ebpFrame;
42924c80d75 (Jonghyun Park   2017-04-05 09:30:08 +0900 2439) #else
42924c80d75 (Jonghyun Park   2017-04-05 09:30:08 +0900 2440)     // For x86/Linux, interruptible code always has full arg info
42924c80d75 (Jonghyun Park   2017-04-05 09:30:08 +0900 2441)     //
42924c80d75 (Jonghyun Park   2017-04-05 09:30:08 +0900 2442)     // This should be aligned with emitFullArgInfo setting at
42924c80d75 (Jonghyun Park   2017-04-05 09:30:08 +0900 2443)     // emitter::emitEndCodeGen (in JIT)
19b1ce56f95 (Jonghyun Park   2017-04-11 07:49:08 +0900 2444)     hasPartialArgInfo = false;
42924c80d75 (Jonghyun Park   2017-04-05 09:30:08 +0900 2445) #endif
```

Reply:
Right, but the logic was the same previously, it was just using `info->ebpFrame` directly instead of `hasPartialArgInfo`